### PR TITLE
refactor(web): spin off method for correction-search probability-thresholding check 📚 

### PR DESCRIPTION
**Pull request by jahorton**
The refactor allows this logic to be unit-tested far more easily.  We additionally leverage this fact and add corresponding unit-tests within this PR.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/lm-worker/src/main/predict-helpers.ts [hunk 1]
```diff
@@ -278,7 +311,7 @@ export async function correctAndEnumerate(
 
     // Only set 'best correction' cost when a correction ACTUALLY YIELDS predictions.
     if(predictions.length > 0 && bestCorrectionCost === undefined) {
-      bestCorrectionCost = -Math.log(predictionRoot.p);
+      bestCorrectionCost = rootCost;
```
Comment (jahorton) [inline common/web/lm-worker/src/main/predict-helpers.ts:314]: Refer to line 307; it's the same value, just without the `exp` / `log` round-trip.

File: common/web/lm-worker/src/main/predict-helpers.ts [hunk 2]
```diff
@@ -13,6 +13,39 @@ import ModelCompositor from './model-compositor.js';
 
 export const AUTOSELECT_PROPORTION_THRESHOLD = .66;
 
+/**
```
Comment (mcdurdin) [inline common/web/lm-worker/src/main/predict-helpers.ts:16]: Loving these comments

File: common/web/lm-worker/src/main/predict-helpers.ts [hunk 3]
```diff
@@ -291,26 +324,8 @@ export async function correctAndEnumerate(
 
     rawPredictions = rawPredictions.concat(predictions);
 
-    let correctionCost = match.totalCost;
-    // Searching a bit longer is permitted when no predictions have been found.
-    if(correctionCost >= bestCorrectionCost + 8) {
+    if(shouldStopSearchingEarly(bestCorrectionCost, match.totalCost, rawPredictions)) {
```
Comment (mcdurdin) [inline common/web/lm-worker/src/main/predict-helpers.ts:327]: :+1: clearer!

**Review by mcdurdin — APPROVED** [review]
LGTM, comments are great, really help with understanding the _why_ for the constants.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.79-alpha](https://keyman.com/downloads/releases/alpha/18.0.79)

**Pull request merged**